aten/src/ATen/functorch/ADInterpreters.cpp

```diff
@@ -44,12 +44,17 @@ Tensor materializeGradWrappers(const Tensor& tensor, int64_t current_level) {
   return makeTensorWrapper(tensor, current_level, /*is_immutable=*/true);
 }
 
+static Tensor base_lift(const Tensor& tensor, int64_t level) {
+  auto tensor_ = unwrapIfDead(tensor);
+  return materializeGradWrappers(tensor_, level);
+}
+
 Tensor GradInterpreterPtr::lift(const Tensor& tensor) const {
-  return materializeGradWrappers(tensor, level());
+  return base_lift(tensor, level());
 }
 
 Tensor JvpInterpreterPtr::lift(const Tensor& tensor) const {
-  return materializeGradWrappers(tensor, level());
+  return base_lift(tensor, level());
```

Comment:
When does this happen? Why didn't we need it before?

Reply:
This is necessary to support functorch.vjp. functorch.vjp constructs a graph using GradWrapperTensors which then all become dead tensor wrappers.

In the C++ the functorch grad transform always unwraps dead tensor wrappers (https://github.com/pytorch/pytorch/blob/a6caa9c54bb5fce93a77e7407fd7e8993a767527/aten/src/ATen/functorch/ADInterpreters.cpp#L60-L68, it happens inside checkForInvalidMutationOnCaptures).

In Python, we want to be consistent with that behavior (https://github.com/pytorch/pytorch/blob/a6caa9c54bb5fce93a77e7407fd7e8993a767527/torch/_functorch/pyfunctorch.py#L107-L110). pyfunctorch today just materializes the wrappers but does not actually unwrap dead wrappers, so this PR fixes it.

Previously we didn't need it because I didn't actually test functorch.vjp; this PR exercises the functorch vjp tests and so we hit this case.